siftseek/endpoints/helpers/query_helpers.py

```diff
@@ -0,0 +1,97 @@
+"""
+query_helpers.py provides helper functions for querying the database in the SiftSeek
+application.
+
+Each helper supports a custom error code for future development.
+"""
+
+from typing import Type, Any, List, Optional, TypeAlias
+
+from flask import abort
+from sqlalchemy import and_
+
+from siftseek.models.db import db, Base
+from siftseek.models.seeker import Seeker
+from siftseek.models.job_application import JobApplication
+
+# TODO Set up TypeAlias for Base
+
+
+def get_model_by_pk_or_404(model: Type[Base], pk: Any) -> Optional[Base]:
```

Comment:
i dont think you should be including http error code logic inside of helper functions like this. Your functions should be as simple and straight forward as possible. You'd typically have the handling the http code in the layer above this. cause returning a 404 when a record never existed makes sense but what if it was created and then deleted and now doesn't exist. Do you still want to return a 404? It depends on your use case at that point.

Reply:
This get_or_404 pattern is addressed here: https://github.com/markalanboyd/SiftSeek-API/pull/27/commits/279647d15ed654e98ed5df541b43c8fa8799fbdd

I was getting an error when trying to use the built-in `get_or_404` before. It said flask-sqlalchemy was using a deprecated sqlalchemy method, so I constructed my own version as a helper to stop the error popping up. When I went back to test it again just now, it seems to work no problem, so I'll refactor to address that.

I see what you mean about the separation of concerns, but the get_or_404 method itself was created to wrap the boilerplate logic you'd have to write every time you wanted to throw a 404 in a get request in flask-sqlalchemy.